src/transaction.rs

```diff
@@ -176,6 +176,31 @@ impl Transaction {
         asset_ids
     }
 
+    /// Convert the type into a JSON string
+    ///
+    /// This is implemented as infallible because serde_json will fail only if the type can't
+    /// serialize one of its attributes. We don't have such case with the transaction because all
+    /// of its attributes are trivially serialized.
+    ///
+    /// If an error happens, a JSON string with the error description will be returned
+    #[cfg(all(feature = "serde", feature = "alloc"))]
+    pub fn to_json(&self) -> alloc::string::String {
+        serde_json::to_string(self).unwrap_or_else(|e| alloc::format!(r#"{{"error": "{}"}}"#, e))
```

Comment:
Does this introduce downstream effects where the function consumer now has to check for the error json?